### PR TITLE
WeekdaysBetween: Convert to full template

### PR DESCRIPTION
**Pull request by rasikapohankar**
###### What does your Pull Request do (check all that apply)?

- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [x] Enhancement: **`WeekdaysBetween: Convert to full template`**

###### Description of changes

Updates WeekdaysBetween Goodie to use full template. Re: #2767 

###### People to notify (@mention interested parties)
@GuiltyDolphin @moollaza 

---

Instant Answer Page: https://duck.co/ia/view/weekdays_between

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @syst3mw0rm

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/WeekdaysBetween.t [hunk 1]
```diff
@@ -9,116 +9,66 @@ use DDG::Test::Goodie;
...
+            title    => $weekday_count,
+            subtitle => "$weekday_plurality between $start_str - $end_str",
+        },
+        templates => {
+            group => "text"
+        }
+      };
+}
+
+sub build_test{ test_zci(build_structured_answer(@_)) }
```
Comment (mintsoft) [inline t/WeekdaysBetween.t]: I'd probably remove `build_structured_answer` and have all this in `build_test`

File: t/WeekdaysBetween.t [hunk 2]
```diff
@@ -9,116 +9,66 @@ use DDG::Test::Goodie;
...
-        input     => ['06 Jan 2014', '10 Jan 2014'],
-        operation => 'Weekdays between',
-        result    => 5,
-    });
+my @jan_6_to_10 = ('06 Jan 2014', '10 Jan 2014', 5);
+
+sub build_structured_answer {
+    my( $start_str, $end_str, $weekday_count) = @_;
+
+    my $verb = $weekday_count == 1 ? 'is' : 'are';
```
Comment (mintsoft) [inline t/WeekdaysBetween.t]: These should be passed as values into `build_test`

File: t/WeekdaysBetween.t [hunk 3]
```diff
@@ -9,116 +9,66 @@ use DDG::Test::Goodie;
...
+    my $weekday_plurality = $weekday_count == 1 ? 'Weekday' : 'Weekdays';
+    my $response = "There $verb $weekday_count $weekday_plurality between $start_str and $end_str.";
+
+    return $response,
```
Comment (mintsoft) [inline t/WeekdaysBetween.t]: Tiny nit, generally we format this as here: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3106/files#diff-1b79e4272ec2bc1488fb1e56a6298135R60
4 spaces per tab

**Comment by mintsoft** [comment]
Thanks @rasikapohankar I've put some minor feedback against the tests to make them have less logic and be more declarative.

**Comment by GuiltyDolphin** [comment]
@rasikapohankar Thanks! :dolphin:

**Comment by rasikapohankar** [comment]
@mintsoft @GuiltyDolphin Thanks!

**Comment by rasikapohankar** [comment]
@GuiltyDolphin Sorry but it looks like I over indented `structured_answer` in the test file, it wasn't looking like that in the editor :disappointed: . Should I submit a pull request to fix it?

**Comment by GuiltyDolphin** [comment]
@rasikapohankar If you want - it isn't a huge issue.

**Comment by rasikapohankar** [comment]
@GuiltyDolphin Thanks a lot! I have submitted a pull request, #3107.

**Pull request merged**